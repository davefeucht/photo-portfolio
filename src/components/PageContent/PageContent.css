.page__content {
    background-color: var(--template-black);
    color: var(--template-white);
    padding: 0 3% 0 5%;
}

.page__content a {
    color: var(--template-blue);
}

.page__content a:visited {
    color: var(--template-transparent-blue);
}