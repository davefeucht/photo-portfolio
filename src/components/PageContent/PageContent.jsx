/** ***************
* PageContent component implements displaying the content of a page in the application
**************** */

import './PageContent.css';

import PropTypes from 'prop-types';
import React from 'react';

function PageContent({ content }) {
    return (
        <div
            className="page__content"
            /* eslint-disable-next-line react/no-danger */
            dangerouslySetInnerHTML={{ __html: content }}
        />
    );
}

PageContent.displayName = 'PageContent';

PageContent.propTypes = {
    content: PropTypes.string.isRequired
};

export default PageContent;
