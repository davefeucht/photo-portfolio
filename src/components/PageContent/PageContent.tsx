/** ***************
* PageContent component implements displaying the content of a page in the application
**************** */

import './PageContent.css';

import * as React from 'react';

interface PageContentProps {
    content: string
}

const PageContent: React.FC<PageContentProps> = ({ content }) => {
    return (
        <div
            className="page__content"
            /* eslint-disable-next-line react/no-danger */
            dangerouslySetInnerHTML={{ __html: content }}
        />
    );
};

PageContent.displayName = 'PageContent';

export default PageContent;
