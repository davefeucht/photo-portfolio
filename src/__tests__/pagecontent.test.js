import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import PageContent from '../components/PageContent/PageContent';

const content = 'This content should be displayed in the page because that is how it works';

test('PageContent displays', () => {
  let component;
  let tree;
  component = renderer.create(
      <MemoryRouter initialEntries={['/page/150']} >
          <PageContent content={content}></PageContent>
      </MemoryRouter>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
