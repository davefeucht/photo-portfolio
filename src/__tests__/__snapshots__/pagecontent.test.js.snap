// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent displays 1`] = `
<div
  className="page__content"
  dangerouslySetInnerHTML={
    {
      "__html": "This content should be displayed in the page because that is how it works",
    }
  }
/>
`;
